src/Allocator/Allocator.hs

```diff
@@ -0,0 +1,64 @@
+module Allocator.Allocator where
+
+import Env
+import Manager
+import Foundation
+import Control.Effect.Error
+import Control.Effect.State
+import Data.HashMap.Lazy
+import Control.Monad
+
+-- map of processes to its pages
+type PMap = HashMap ProcessId [Page 'Ram]
+type Ptr = Int
+newtype Size = Size { unSize :: CountOf Word8 } deriving (Eq, Show)
+
+data AllocError = OutOfMemory deriving (Eq, Ord)
+
+type AllocSig sig m = ( Has (State PMap) sig m
+                      , Has (Throw AllocError) sig m
```

Comment:
Try using `stylish-haskell` pretty printer(you can choose it in vscode, and then press `Ctrl+Shift+I` in the file)